##ResNet50

In [2]:
from tensorflow.keras.applications.resnet50 import ResNet50
from tensorflow.keras.applications.resnet50 import preprocess_input, decode_predictions
from tensorflow.keras.preprocessing import image
import numpy as np

2024-10-06 16:23:31.385912: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-10-06 16:23:31.470113: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-10-06 16:23:31.493342: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-10-06 16:23:31.641609: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-10-06 16:23:32.957884: W tensorflow/compiler/tf2

In [3]:
model = ResNet50(weights="imagenet")

102967424/102967424 ━━━━━━━━━━━━━━━━━━━━ 51s 0us/step


In [10]:
test_path = "cabbage_butterfly.jpg"
img = image.load_img(test_path, target_size=(224,224))
img = image.img_to_array(img)
img = np.expand_dims(img, axis=0) #4boyutlu
img = preprocess_input(img)

prediction = model.predict(img)

1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 53ms/step


In [11]:
prediction

array([[4.21878221e-09, 1.38828826e-09, 5.19764544e-11, 2.15476678e-10,
        4.48598624e-11, 1.10949303e-08, 1.20176824e-09, 3.41984645e-08,
        1.09407061e-08, 5.47287271e-10, 5.54703199e-08, 5.90490004e-07,
        5.07766593e-08, 1.54828754e-08, 9.52060134e-07, 1.63942424e-08,
        4.46662312e-07, 8.79315394e-08, 3.49943519e-09, 2.65017235e-07,
        2.37343976e-08, 1.11714185e-08, 1.02248099e-09, 1.69404091e-09,
        1.64267959e-08, 4.84465934e-09, 5.03491400e-08, 2.88457667e-08,
        2.30910202e-09, 2.53710697e-09, 1.33569722e-09, 5.53756081e-08,
        1.54071877e-09, 3.73704934e-09, 2.15513363e-09, 6.51483978e-09,
        1.78710220e-08, 4.48800153e-09, 8.76946693e-09, 4.72466155e-08,
        2.62022610e-07, 4.37740511e-08, 7.47357873e-08, 1.59785909e-08,
        1.62970650e-08, 2.76840240e-09, 5.34645643e-08, 3.68995217e-07,
        2.29153430e-09, 9.42160239e-10, 1.12041043e-09, 3.66895597e-10,
        1.26185942e-08, 1.47907442e-09, 3.18926352e-09, 6.372899

In [12]:
prediction = decode_predictions(prediction, top=3)[0]
prediction

[('n02280649', 'cabbage_butterfly', 0.782488),
 ('n02281406', 'sulphur_butterfly', 0.20118274),
 ('n02277742', 'ringlet', 0.015934054)]

##VGG16

In [1]:
from tensorflow.keras.applications.vgg16 import VGG16
from tensorflow.keras.applications.vgg16 import preprocess_input, decode_predictions
from tensorflow.keras.preprocessing import image
import numpy as np

2024-10-04 19:24:28.202261: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-10-04 19:24:28.220736: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-10-04 19:24:28.226017: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-10-04 19:24:28.238491: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-10-04 19:24:29.143275: W tensorflow/compiler/tf2

In [3]:
model = VGG16(weights = "imagenet")

553467096/553467096 ━━━━━━━━━━━━━━━━━━━━ 354s 1us/step


In [6]:
test_path = "cabbage_butterfly.jpg"
img = image.load_img(test_path, target_size=(224,224))
img = image.img_to_array(img)
img = np.expand_dims(img, axis=0)
img = preprocess_input(img)

prediction = model.predict(img)
prediction


1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 245ms/step


array([[9.32275812e-10, 6.18950668e-10, 6.91257260e-11, 6.47365606e-10,
        4.23813033e-11, 9.67326841e-09, 1.31225686e-09, 7.40434247e-09,
        4.08317380e-09, 2.92725774e-11, 4.39483339e-09, 2.92551960e-07,
        2.66091895e-08, 6.52586749e-08, 3.31651819e-07, 4.94182828e-09,
        2.39522251e-07, 5.78340718e-08, 1.57793756e-09, 4.86938276e-08,
        1.26654753e-09, 4.65383687e-09, 3.00735020e-10, 1.74041115e-10,
        1.03670796e-08, 3.82759241e-10, 9.13259690e-09, 1.42020384e-09,
        4.34731806e-10, 1.26350486e-09, 4.45798981e-10, 2.04344914e-08,
        3.50928231e-10, 2.83973622e-10, 1.09400433e-09, 1.22001242e-09,
        1.38785006e-09, 1.53166271e-10, 1.39384282e-09, 1.97069099e-08,
        6.08231062e-07, 2.43743514e-09, 8.39293968e-09, 6.92104951e-09,
        2.39042883e-08, 7.18526488e-11, 2.24142070e-07, 1.50013733e-07,
        2.14494811e-10, 8.31564886e-11, 3.87948285e-10, 5.71218316e-11,
        6.73271883e-10, 1.28151559e-10, 2.11365217e-10, 1.008752

In [7]:
prediction = decode_predictions(prediction, top=3)[0]
prediction

[('n02280649', 'cabbage_butterfly', 0.83566654),
 ('n02281406', 'sulphur_butterfly', 0.10747166),
 ('n02277742', 'ringlet', 0.04398086)]

##VGG19

In [4]:
from tensorflow.keras.applications.vgg19 import VGG19
from tensorflow.keras.applications.vgg19 import preprocess_input, decode_predictions
from tensorflow.keras.preprocessing import image
import numpy as np

In [5]:
model = VGG19(weights = "imagenet")

A local file was found, but it seems to be incomplete or outdated because the auto file hash does not match the original value of cbe5617147190e668d6c5d5026f83318 so we will re-download the data.
574710816/574710816 ━━━━━━━━━━━━━━━━━━━━ 136s 0us/step


In [6]:
test_path = "cabbage_butterfly.jpg"
img = image.load_img(test_path, target_size=(224,224))
img = image.img_to_array(img)
img = np.expand_dims(img, axis=0)
img = preprocess_input(img)

prediction = model.predict(img)
prediction


1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 318ms/step


array([[4.28163771e-09, 4.44775061e-09, 5.23009101e-11, 5.63784797e-10,
        3.75776424e-11, 7.93180188e-09, 3.18376103e-09, 6.52535581e-09,
        6.43303011e-09, 1.67329067e-10, 4.19573496e-08, 1.09135328e-06,
        4.83558402e-08, 1.23048252e-07, 1.63511538e-06, 1.95480379e-08,
        7.38246229e-08, 1.22486625e-07, 3.33334853e-08, 6.60782149e-08,
        1.77774524e-08, 1.34580853e-08, 1.56314872e-09, 4.55335436e-10,
        3.67024526e-08, 4.10901979e-09, 3.72512190e-08, 1.04148068e-08,
        8.60312155e-10, 1.15101466e-08, 5.35989741e-09, 7.05036243e-08,
        7.94527732e-09, 8.13336454e-10, 4.46570381e-09, 1.36191070e-09,
        1.08901084e-08, 3.00104802e-10, 8.23413615e-10, 4.47865460e-08,
        5.10829352e-07, 1.23630315e-08, 1.01457076e-07, 1.77989286e-08,
        1.62649467e-08, 2.27085184e-10, 3.64770500e-07, 2.45743649e-07,
        3.88616417e-10, 4.02454153e-10, 3.50710372e-09, 2.19426456e-11,
        3.61593400e-09, 2.40404391e-10, 1.57455859e-09, 1.646411

In [7]:
prediction = decode_predictions(prediction, top=3)[0]
prediction

[('n02280649', 'cabbage_butterfly', 0.9452888),
 ('n02281406', 'sulphur_butterfly', 0.036120933),
 ('n02277742', 'ringlet', 0.009066579)]

## ResNet50---> 'cabbage_butterfly', 0.782488
## VGG16 -----> 'cabbage_butterfly', 0.83566654
## VGG19 -----> 'cabbage_butterfly', 0.9452888